$NetBSD: patch-ab,v 1.2 2021/12/29 09:36:37 dholland Exp $

Don't use sgtty, or tparm.

Don't redefine memcpy/memmove; it's completely unnecessary and breaks
SSP.

--- tads2/osunixt.h.orig	2004-02-17 02:20:41.000000000 +0000
+++ tads2/osunixt.h
@@ -84,12 +84,12 @@ Tue Nov 22 15:16:10 EST 1994    Dave Bag
  * Define the following to use sgtty.h instead of termios or direct
  * ioctl hacking.
  */
-#if defined(NEXT) || defined(IBM_RT) || defined(FREEBSD_386) || defined(IBM_AIX) || defined(NETBSD) || defined(OPENBSD) || defined(DARWIN)
+#if 0
 #define USE_IOCTL_INSTEAD_OF_TERMIOS
 #define USE_SGTTY
 #endif
 
-#if !defined(SUN_SPARC_SUNOS) && !defined(SUN3) && !defined(ULTRIX_MIPS) && !defined(LINUX_386) && !defined(NEXT) && !defined(IBM_RT) && !defined(IBM_AIX)
+#if !defined(SUN_SPARC_SUNOS) && !defined(SUN3) && !defined(ULTRIX_MIPS) && !defined(LINUX_386) && !defined(NEXT) && !defined(IBM_RT) && !defined(IBM_AIX) && !defined(__NetBSD__)
 #define HAVE_TPARM      /* define if this system has the tparm routine */
 #endif
 
@@ -160,9 +160,11 @@ FILE* our_fopen(char *filename, char *fl
  * Some machines are missing memmove, so we use our own memcpy/memmove 
  * routine instead.
  */
+#if !defined(__STDC__)
 void *our_memcpy(void *dst, const void *src, size_t size); 
 #define memcpy our_memcpy
 #define memmove our_memcpy
+#endif
 
 /* display lines on which errors occur */
 /* #  define OS_ERRLINE 1 */
